### PR TITLE
HLR & SC | Set up radio button tracking

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > Analytics for form system radio buttons was recently added (by adding `enableAnalytics: true` to the `ui:options`) in our Notice of Disagreement form. We also noted that our other appeals forms - Higher-Level Review and Supplemental Claim - needed analytics added to all radio buttons.
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits decision reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#60431](https://github.com/department-of-veterans-affairs/va.gov-team/issues/60431)

## Testing done

- _Describe what the old behavior was prior to the change_
  > No GA events recorded
- _Describe the steps required to verify your changes are working as expected_
  > Test checking radio selections in staging or in a review instance
- _Describe the tests completed and the results_
  > Updated component & page unit tests; all passing
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

| GA events |
| ------- |
| HLR |
| homeless event |
| informal conference event |
| informal conference time event |
| SC |
| primary phone event |
| Include VA evidence event |
| Private evidence event |
| upload evidence event |

## What areas of the site does it impact?

HLR & Supplemental Claims

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by data-doge — APPROVED** [review]
LGTM. I was able to reproduce the events shown in your screenshots.

Out of curiosity: why is the "Choose the best time for us to call" label accompanied by this object?
```json
{"key":null,"ref":null,"props":{"children":[{"type":"span","key":null,"ref":null,"props":{"className":"contact-choice selected-rep","children":"your representative"},"_owner":null,"_store":{}},{"type":"span","key":null,"ref":null,"props":{"className":"contact-choice selected-me","children":"you"},"_owner":null,"_store":{}}]},"_owner":null,"_store":{}}
```

File: src/applications/appeals/995/pages/evidencePrivateRequest.js [hunk 1]
```diff
@@ -1,22 +1,8 @@
...
-      forceDivWrapper: true,
-    },
-    [EVIDENCE_PRIVATE]: {
-      'ui:title': privateRecordsRequestTitle,
-      'ui:options': {
-        forceDivWrapper: true,
-        hideOnReview: true,
-      },
-    },
+    [EVIDENCE_PRIVATE]: {},
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/995/pages/evidencePrivateRequest.js:5]: We're using `CustomPage` which doesn't require any `uiSchema`, but if you have any required fields that need validation on the review & submit page, then you need to at least add a `ui:validation` entry. This is a required yes/no question, but only needs validation on the page, not the review & submit page.

**Comment by Mottie** [comment]
The radio widget component is receiving the label as JSX, so it [looks in the child props by default](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/widgets/RadioWidget.jsx#L39-L40).

```js
// title may be a React component
const title = options.title?.props?.children || options.title || '';
```

In this case, the "Choose the best time for us to call" label has nested children, so the `title` ends up being an array with more nested children. I know it looks bad here, but I don't think GA cares? Also, I plan on fixing this label once we start working on the form update.

File: src/applications/appeals/995/pages/evidencePrivateRequest.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/995/pages/evidencePrivateRequest.js:5]: So apparently, removing the `hideOnReview` ui option caused that page to show up on the review & submit page even though the `CustomPageReview` is set to `null` - so I'll need to add this option back

**Pull request merged**